src/libs/battle_interface/src/sea/ibattle.cpp

```diff
@@ -75,9 +75,6 @@ BATTLE_INTERFACE::~BATTLE_INTERFACE()
 {
     // STORM_DELETE(m_pMessageIcons);
     STORM_DELETE(m_pShipIcon);
-    m_TextArray.clear();
```

Comment:
what is the other place where they get destructed?

Reply:
In their destructors :)

Reply:
`BILinesInfo::~BILinesInfo()
{
    Release();
}`
`BIImagesInfo::~BIImagesInfo()
{
    Release();
}`
The erased code is in `BATTLE_INTERFACE` class that contains both, so it will release resources again. Since `m_ImagesInfo` not nulling pointers after delete, so it will lead to a crash.
Btw, I have no idea how this worked before. This was probably unused in COAS/TEHO, but I hadn't check specifically.

Reply:
ah, makes sense.